webpack.config.js

```diff
@@ -93,6 +94,13 @@ module.exports = {
 	entry: gutenbergPackages.reduce( ( memo, packageName ) => {
 		const name = camelCaseDash( packageName );
 		memo[ name ] = `./packages/${ packageName }`;
+
+		const modalsPath = `${ memo[ name ] }/src/navigation/modals`;
```

Comment:
Maybe we should do something a bit more generic. Basically add `packages/block-library/src/*/frontend.js` as entry points, so that any block that has a "frontend.js" on its folder get that frontend script automatically registered and enqueued properly.

Reply:
Agreed, that is what I was thinking; creating a standard directory, or file, that if present will be picked up by webpack and output to `build/` so that we can use it later.

I'll add that generic behavior here, so that we can see it working.

Reply:
It would be important to come up with a good pattern here as we will have to promote it in official docs when adding support to `@wordpress/scripts`. This is how it's handled for CSS at the moment:

https://github.com/WordPress/gutenberg/tree/trunk/packages/scripts#using-css

> `index.css` – all imported CSS files are bundled into one chunk named after the entry point, which defaults to index.js, and thus the file created becomes index.css. This is for styles used only in the editor.
> `style-index.css` – imported style.css file(s) (applies to SASS and SCSS extensions) get bundled into one style-index.css file that is meant to be used both on the front-end and in the editor.

There is also `index.js` in the `build` folder, so it feels like in `@wordpress/scripts` it should be `script-index.js`.

Although, maybe we don't need to care about it for core blocks. Yes, I guess it's fine to pick whatever makes sense. It probably would be better to go with `script.js` if we were to continue using `script` from `register_block_type`.

Reply:
> Maybe we should do something a bit more generic. Basically add `packages/block-library/src/*/frontend.js` as entry points, so that any block that has a "frontend.js" on its folder get that frontend script automatically registered and enqueued properly.

Added something like that in https://github.com/WordPress/gutenberg/pull/29537/commits/828e5024f862b1f2c5ac4d59883f86054c0ca5ee

Right now it:
- Looks for `script.js` files within all blocks in the block-library package.
- Builds that file and puts it in `./build/block-library/blocks/<block-name>/script.js`, and
- The `block.json` file at that location takes care of pulling that JS everywhere; editor and frontend.